packages/flutter/test/widgets/list_view_semantics_test.dart

```diff
@@ -0,0 +1,137 @@
+
+      semantics.dispose();
+    });
+
+    testWidgets('reverse vertical', (WidgetTester tester) async {
+      final SemanticsTester semantics = SemanticsTester(tester);
+      final ScrollController controller = ScrollController();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: ListView.builder(
+            reverse: true,
+            controller: controller,
+            itemCount: 10,
+            itemBuilder: (BuildContext context, int index) {
+              return SizedBox(
+                height: 150,
+                child: Text('Tile $index'),
+              );
+            },
+          ),
+        ),
+      );
+
+      expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.scrollDown]));
+
+      // Jump to the end.
+      controller.jumpTo(10 * 150.0);
+      await tester.pumpAndSettle();
+      expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.scrollUp]));
+
+      semantics.dispose();
+    });
+
+    testWidgets('forward horizontal', (WidgetTester tester) async {
+      final SemanticsTester semantics = SemanticsTester(tester);
+      final ScrollController controller = ScrollController();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: ListView.builder(
+            scrollDirection: Axis.horizontal,
+            controller: controller,
+            itemCount: 10,
+            itemBuilder: (BuildContext context, int index) {
+              return SizedBox(
+                height: 150,
+                child: Text('Tile $index'),
+              );
+            },
+          ),
+        ),
+      );
+
+      expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.scrollLeft]));
+
+      // Jump to the end.
+      controller.jumpTo(10 * 150.0);
+      await tester.pumpAndSettle();
+      expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.scrollRight]));
+
+      semantics.dispose();
+    });
+
+    testWidgets('reverse horizontal', (WidgetTester tester) async {
+      final SemanticsTester semantics = SemanticsTester(tester);
+      final ScrollController controller = ScrollController();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: ListView.builder(
+            scrollDirection: Axis.horizontal,
+            reverse: true,
+            controller: controller,
+            itemCount: 10,
+            itemBuilder: (BuildContext context, int index) {
+              return SizedBox(
+                height: 150,
+                child: Text('Tile $index'),
+              );
+            },
+          ),
+        ),
+      );
+
+      expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.scrollRight]));
+
+      // Jump to the end.
+      controller.jumpTo(10 * 150.0);
```

Comment:
Nit: Declare constants for 10 and 150?  It's pretty obvious what they are in these short tests, not a big deal.

Reply:
Done.